prow/plugins/blunderbuss/blunderbuss.go

```diff
@@ -109,63 +117,86 @@ func handlePullRequest(pc plugins.Agent, pre github.PullRequestEvent) error {
 	return handle(
 		pc.GitHubClient,
 		oc, pc.Logger,
-		pc.PluginConfig.Blunderbuss.ReviewerCount,
-		pc.PluginConfig.Blunderbuss.FileWeightCount,
-		pc.PluginConfig.Blunderbuss.MaxReviewerCount,
-		pc.PluginConfig.Blunderbuss.ExcludeApprovers,
-		&pre,
+		pc.PluginConfig.Blunderbuss,
+		&pre.Repo,
+		&pre.PullRequest,
+	)
+}
+
+func handleGenericCommentEvent(pc plugins.Agent, ce github.GenericCommentEvent) error {
+	if ce.Action != github.GenericCommentActionCreated || !ce.IsPR || ce.IssueState == "closed" {
+		return nil
+	}
+
+	pr, err := pc.GitHubClient.GetPullRequest(ce.Repo.Owner.Login, ce.Repo.Name, ce.Number)
+	if err != nil {
+		return err
+	}
+
+	ro, err := pc.OwnersClient.LoadRepoOwners(ce.Repo.Owner.Login, ce.Repo.Name, pr.Base.Ref)
```

Comment:
Can we DRY this out as well?

Reply:
This will require more changes compared to changing the `handle` method signature.

We'll need to create a mock of `repoowners.Interface` in this package to pass that to `handle` and call `repoowners.Interface.LoadRepoOwners` once there instead of doing it in the other two functions.

I started working on this but wanted to confirm before I continue. :grin:

Reply:
Or just factor out a smaller func that resolves the owners client and calls `handle`?

Reply:
Merged this with the changes needed to write more isolated tests.